src/CLICommand/parameters_helper.ts

```diff
@@ -216,14 +221,21 @@ export class ParametersHelper {
 
 	/**
 	 * @param {ParameterName} parameterName
+	 * @param {(input: any) => T} mapFunc A function that maps the parameter value into a T instance
 	 * @returns {string | undefined}
-	 * Returns the string value for the specific parameter; throws an error if not set
+	 * @throws - When the required parameter value has a falsy value
+	 * Returns the string value for the specific parameter
 	 */
-	public getRequiredParameterValue(parameterName: ParameterName): string {
+	public getRequiredParameterValue<T = string>(
+		parameterName: ParameterName,
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		mapFunc: (input: any) => T = (input: any) => input as T
+	): T {
+		// FIXME: it could also return an array or a boolean!
 		const value = this.options[parameterName];
 		if (!value) {
 			throw new Error(`Required parameter ${parameterName} wasn't provided!`);
 		}
```

Comment:
Shouldn't this work without the anys? 

```ts
	public getRequiredParameterValue<T = string>(
		parameterName: ParameterName,
		mapFunc: (value: T) => T = (value: T) => value as T
	): T {
		const value = this.options[parameterName];
		if (value === undefined) {
			throw new Error(`Required parameter ${parameterName} wasn't provided!`);
		}
		return mapFunc(value);
```

And I don't see why boolean/arrays aren't okay here, we have a generic. We should be able to call this on a required boolean like so:

```ts
const dryRun = parameters.getRequiresParameterValue<boolean>(DryRunParameter);
```

I might be missing something with how they're returned by commander though

Reply:
The input value isn't a T. It's whatever options.whatever is, and they are seen by the compiler as `any`

Reply:
The options type in this file doesn't match the options type from the action class. It should have the type from `export type ParsedParameters = Record<ParameterName, unknown>;`

And when we add the generic, we know exactly what unknown is (or should be). So the options in this function would be `Record<ParameterName, T>` rather than any.

I think it could be easily represented by casting `const value: T = this.options[parameterName]`